ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -271,15 +271,17 @@ public void setAutoCapitalize(ReactEditText view, int autoCapitalize) {
 
   @ReactProp(name = "keyboardType")
   public void setKeyboardType(ReactEditText view, @Nullable String keyboardType) {
+    int flagsToUnset = 0;
     int flagsToSet = 0;
     if (KEYBOARD_TYPE_NUMERIC.equalsIgnoreCase(keyboardType)) {
-      flagsToSet = InputType.TYPE_CLASS_NUMBER;
+      flagsToUnset = InputType.TYPE_MASK_CLASS | InputType.TYPE_MASK_FLAGS;
```

Comment:
Why would we need to clear all these flags? It looks like we are only using InputType.TYPE_CLASS_NUMBER, InputType.TYPE_NUMBER_FLAG_DECIMAL, InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS. I much prefer the old version of this implementation, where we unset all flags that we could be setting, and then set the ones that we actually need